crates/kitsune_p2p/proxy/src/tx2.rs

```diff
@@ -144,7 +156,57 @@ fn promote_addr(base_addr: &TxUrl, cert: &Tx2Cert) -> KitsuneResult<TxUrl> {
         .into())
 }
 
-#[allow(dead_code)]
+#[derive(Clone)]
+struct Backoff {
+    bo: Arc<AtomicUsize>,
+    n: Arc<Notify>,
+    init: usize,
+    max: usize,
+}
+
+impl Backoff {
+    pub fn new(init: usize, max: usize) -> Self {
+        Self {
+            bo: Arc::new(AtomicUsize::new(init)),
+            n: Arc::new(Notify::new()),
+            init,
+            max,
+        }
+    }
+
+    pub fn reset(&self) {
+        self.bo.store(self.init, Ordering::SeqCst);
+        self.n.notify_waiters();
```

Comment:
What happens if `reset` is called just before a wait is called? Would that be a problem that the waiter won't be notified?

Reply:
I think that condition might be unavoidable. At least, the timing should be reset, so it'll only wait 10 ms before checking the proxy connection is active again.